e3sm_diags/logger.py

```diff
@@ -91,49 +93,22 @@ def custom_logger(name: str, propagate: bool = True) -> logging.Logger:
     return logger
 
 
-def _update_root_logger_filepath_to_prov_dir(log_path: str):
-    """Updates the log file path to the provenance directory.
+def _add_filehandler(log_path: str):
+    """Adds a file handler to the root logger dynamically.
 
-    This method changes the log file path to a subdirectory named 'prov'
-    within the given results directory. It updates the filename of the
-    existing file handler to the new path.
+    Adding the file handler will also create the log file automatically.
 
     Parameters
     ----------
     log_path : str
-        The path to the log file, which is stored in the `results_dir`
-        sub-directory called "prov".
+        The path to the log file.
 
     Notes
     -----
-    - The method assumes that a logging file handler is already configured.
-    - The log file is closed and reopened at the new location.
-    - The log file mode is determined by the constant `LOG_FILEMODE`.
-    - The log file name is determined by the constant `LOG_FILENAME`.
+    Any warnings that appear before the log filehandler is instantiated will not
+    be captured (e.g,. esmpy VersionWarning). However, they will still be
+    captured by the console via the default StreamHandler.
     """
-    for handler in logging.root.handlers:
-        if isinstance(handler, logging.FileHandler):
-            handler.baseFilename = log_path
-            handler.stream.close()
-            handler.stream = open(log_path, LOG_FILEMODE)  # type: ignore
-            break
-
-
-def move_log_to_prov_dir(results_dir: str, logger: logging.Logger):
-    """Moves the e3sm diags log file to the provenance directory.
-
-    This function should be called at the end of the diagnostic run to capture
-    all console outputs.
-
-    Parameters
-    ----------
-    results_dir : str
-        The results directory for the run.
-    """
-    provenance_dir = f"{results_dir}/prov/{LOG_FILENAME}"
-
-    # Must copy and then delete because shutil.move does not work if different
-    # filesystems are used for the source and destination directories.
-    shutil.copy(LOG_FILENAME, provenance_dir)
-    os.remove(LOG_FILENAME)
-    logger.info(f"Log file saved in {provenance_dir}")
+    file_handler = logging.FileHandler(log_path, mode=LOG_FILEMODE)
+    file_handler.setFormatter(logging.Formatter(LOG_FORMAT))
+    logging.root.addHandler(file_handler)
```

Comment:
Instead of changing the path of the root logger's `FileHandler`, we call `_add_filehandler()` later to add a `FileHandler` that saves the log file in the `prov` directory.

This prevents the creation of a duplicate `e3sm_diags_run.log` that is stored in the current working directory, where `e3sm_diags` is invoked.